### PR TITLE
fix: ID-2067 Remove use of Buffer in personal sign

### PR DESCRIPTION
**Pull request by imx-mikhala**
### Hi👋, please prefix this PR's title with:

- [ ] `breaking-change:` if you have introduced modification that necessitates immediate adjustments by this SDK's users to their applications, clients, or integrations to avert disruptions to existing features or functionalities.
- [x] `feat:`, `fix:`, `refactor:`, `docs:`, or `chore:`.

# Summary
- Removes Buffer to fix personal sign when Buffer is not defined

# Detail and impact of the change

[Here](https://github.com/imx-mikhala/passport-browser-bundle-sdk) is a sample-app put together to test Passport using browser bundle

There is an SDK alpha release of this branch [here](https://cdn.jsdelivr.net/npm/@imtbl/sdk@1.45.0-alpha.4/dist/index.browser.min.js)

When Buffer was missing, the confirmation app displays the hex string of the message:

telegram-cloud-photo-size-5-6332289006158593993-y

The following is with the patch applied:

Screenshot 2024-06-27 at 1 23 13 PM

### CODE REVIEW

File: packages/passport/sdk/src/zkEvm/personalSign.ts [hunk 1]
```diff
@@ -38,7 +38,12 @@ export const personalSign = async ({
     throw new JsonRpcError(RpcErrorCode.INVALID_PARAMS, 'personal_sign requires the signer to be the from address');
   }
 
-  // Convert message into a string if it's a hex
+  if (typeof window !== 'undefined' && !window.Buffer) {
```
Comment (keithbro-imx) [inline packages/passport/sdk/src/zkEvm/personalSign.ts:41]: 👀 very interesting.. why does this work?

Comment (imx-mikhala) [reply packages/passport/sdk/src/zkEvm/personalSign.ts:41]: The `hexToString` function called after this if statement uses `Buffer` which causes an issue with using the browser bundle as `Buffer` doesn't exist. Adding it to the window if Buffer isnt defined fixes the issue.

File: packages/passport/sdk/src/utils/string.ts [hunk 2]
```diff
@@ -1,18 +1,11 @@
...
-    // Ensures Buffer is available when using the browser bundle of the SDK
-    if (typeof window !== 'undefined' && !window.Buffer) {
-      const bufferModule = await import('buffer');
-      window.Buffer = bufferModule.Buffer;
-    }
-
     const stripped = utils.stripZeros(utils.arrayify(hex));
-    const buff = Buffer.from(stripped);
-    return buff.length === 32 ? hex : utils.toUtf8String(stripped);
+    return utils.toUtf8String(stripped);
```
Comment (imx-mikhala) [inline packages/passport/sdk/src/utils/string.ts:8]: todo: confirm purpose of `buff.length === 32 `

File: packages/passport/sdk/src/utils/string.ts [hunk 3]
```diff
@@ -5,8 +5,7 @@ export const hexToString = (hex: string) => {
 
   try {
     const stripped = utils.stripZeros(utils.arrayify(hex));
-    const buff = Buffer.from(stripped);
-    return buff.length === 32 ? hex : utils.toUtf8String(stripped);
+    return stripped.length === 32 ? hex : utils.toUtf8String(stripped);
```
Comment (imx-mikhala) [inline packages/passport/sdk/src/utils/string.ts:8]: @Jon-Alonso @haydenfowler  Can someone confirm what the `length === 32` check is for here?

Comment (imx-mikhala) [reply packages/passport/sdk/src/utils/string.ts:8]: Thanks for the info. We can probably just remove the check then if other implementations result in the same behaviour.

File: packages/passport/sdk/src/utils/string.ts [hunk 4]
```diff
@@ -5,8 +5,7 @@ export const hexToString = (hex: string) => {
 
   try {
     const stripped = utils.stripZeros(utils.arrayify(hex));
-    const buff = Buffer.from(stripped);
```
Comment (keithbro-imx) [inline packages/passport/sdk/src/utils/string.ts:8]: Love it when the fix to a bug is deleting code 😂

**Review by keithbro-imx — APPROVED** [review]

**Pull request merged**